### PR TITLE
Added new CreateUser, Update, Delete Procedure Using JSON

### PR DESCRIPTION
**Pull request by Abruna-Afrin**
Added new CreateUser, Update, Delete Procedure Using JSON_V3

### CODE REVIEW

File: src/database/Matrimonial/migrations/V003_20241006_Add CreateUser and delete and update.sql [hunk 1]
```diff
@@ -1,104 +1,162 @@
 DELIMITER //
-CREATE PROCEDURE CreateUser(
-    IN p_FirstName VARCHAR(255),
+CREATE PROCEDURE CreateUsers(
```
Comment (yousufhossain) [inline src/database/Matrimonial/migrations/V003_20241006_Add CreateUser and delete and update.sql]: rename to `CreateUser`

File: src/database/Matrimonial/migrations/V003_20241006_Add CreateUser and delete and update.sql [hunk 2]
```diff
@@ -1,104 +1,162 @@
...
+    
+    SELECT last_insert_id() INTO p_Errors;
+    
 
-    -- Return new user ID
-    SET p_UserId = LAST_INSERT_ID();
 END //
 
-CREATE PROCEDURE UpdateUser(
+CREATE  PROCEDURE Update_Users(
```
Comment (yousufhossain) [inline src/database/Matrimonial/migrations/V003_20241006_Add CreateUser and delete and update.sql]: rename `UpdateUser`

File: src/database/Matrimonial/migrations/V003_20241006_Add CreateUser and delete and update.sql [hunk 3]
```diff
@@ -1,104 +1,162 @@
...
+    
+    IF  p_Gender!= 'Male' AND  p_Gender != 'Female' THEN
+    SET @p_ErrorTable = JSON_ARRAY_APPEND(@p_ErrorTable, '$', 'Gender cant be null or empty');
+    END IF;
+    
+    IF p_Email IS NULL OR p_Email = '' OR NOT p_Email REGEXP '^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,4}$' THEN
+    SET @p_ErrorTable = JSON_ARRAY_APPEND(@p_ErrorTable, '$', 'Invalid Email ');
+    END IF;
+    
+    IF p_Phone IS NULL OR p_Phone = '' THEN
```
Comment (yousufhossain) [inline src/database/Matrimonial/migrations/V003_20241006_Add CreateUser and delete and update.sql:101]: we can use regex here as well

https://stackoverflow.com/questions/2113908/what-regular-expression-will-match-valid-international-phone-numbers

\+(9[976]\d|8[987530]\d|6[987]\d|5[90]\d|42\d|3[875]\d|
2[98654321]\d|9[8543210]|8[6421]|6[6543210]|5[87654321]|
4[987654310]|3[9643210]|2[70]|7|1)\d{1,14}$

File: src/database/Matrimonial/migrations/V003_20241006_Add CreateUser and delete and update.sql (hunk 1 above)
Comment (Abruna-Afrin) [reply src/database/Matrimonial/migrations/V003_20241006_Add CreateUser and delete and update.sql]: done

File: src/database/Matrimonial/migrations/V003_20241006_Add CreateUser and delete and update.sql (hunk 2 above)
Comment (Abruna-Afrin) [reply src/database/Matrimonial/migrations/V003_20241006_Add CreateUser and delete and update.sql]: done

**Pull request merged**